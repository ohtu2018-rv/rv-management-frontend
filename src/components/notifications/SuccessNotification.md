### With shadow

```js
<div style={{width: 250}}><SuccessNotification message={"This is a success message"} shadow /></div>
```

### Without shadow

```js
<div style={{width: 250}}><SuccessNotification message={"This is a success message"} /></div>
```
