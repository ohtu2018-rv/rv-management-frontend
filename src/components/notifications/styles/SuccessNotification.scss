@import "vars";

.success-message {
  font-size: $notification-font-size;
  padding: $base-padding;
  text-align: center;
  display: block;
  white-space: nowrap;
  background-color: $success-notification-bgcolor;
  border-radius: $notification-border-radius;
  color: $success-notification-textcolor;
  margin: $base-margin;
}

.success-message-shadow {
  box-shadow: $box-shadow;
}

#successAnimation {
  display: inline-block;
  margin-left: auto;
  margin-right: auto;
}

#successAnimationCircle {
  stroke-dasharray: 151px 151px;
  stroke: $success-notification-textcolor;
}

#successAnimationCheck {
  stroke-dasharray: 36px 36px;
  stroke: $success-notification-textcolor;
}

#successAnimationResult {
  fill: none;
  opacity: 0;
}
