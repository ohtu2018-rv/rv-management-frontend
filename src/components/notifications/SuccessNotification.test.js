import React from 'react';
import { shallow } from 'enzyme';
import SuccessNotification from './SuccessNotification';
import renderer from 'react-test-renderer';

describe.only('<SuccessNotification />', () => {
    it('renders correctly', () => {
        const cmpnt = renderer
            .create(<SuccessNotification message="Test notification" />)
            .toJSON();
        expect(cmpnt).toMatchSnapshot();
    });
});
