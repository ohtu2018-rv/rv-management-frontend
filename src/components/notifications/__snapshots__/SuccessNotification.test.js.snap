// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessNotification /> renders correctly 1`] = `
<div
  className="success-message"
>
  <div
    className="inner"
  >
    <svg
      className="animated"
      height="70"
      id="successAnimation"
      viewBox="0 0 70 70"
      width="70"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M35,60 C21.1928813,60 10,48.8071187 10,35 C10,21.1928813 21.1928813,10 35,10 C48.8071187,10 60,21.1928813 60,35 C60,48.8071187 48.8071187,60 35,60 Z M23.6332378,33.2260427 L22.3667622,34.7739573 L34.1433655,44.40936 L47.776114,27.6305926 L46.223886,26.3694074 L33.8566345,41.59064 L23.6332378,33.2260427 Z"
        fill="#D8D8D8"
        id="successAnimationResult"
      />
      <circle
        cx="35"
        cy="35"
        fill="transparent"
        id="successAnimationCircle"
        r="24"
        stroke="#979797"
        strokeLinecap="round"
        strokeWidth="2"
      />
      <polyline
        fill="transparent"
        id="successAnimationCheck"
        points="23 34 34 43 47 27"
        stroke="#979797"
        strokeWidth="2"
      />
    </svg>
    <div
      className="notification-text"
    >
      Test notification
    </div>
  </div>
</div>
`;
